### ISSUE TITLE
Incrust additional things (images, files, etc.)

### ISSUE THREAD

**Issue by riahc3** [pglombardo/PasswordPusher#2544]
## 🚀 Feature Request

Right now, only text based passwords can be passed. Other things, such as images with QR Codes, files such as certificates used as authentication, etc. might be interesting.

## 🔈 Motivation

Be more than just pass passwords...Pass credentials

**Comment by pglombardo** [comment]
Hi @riahc3 - I agree/good points.  

I have QR codes planned - you enter the content (text, url, otp code), a QR code is generated and sent to the end user to scan with his/her phone.

As for certificates - that could be interesting but I would need to know more specifics.  Right now agnostic file attachments are the only way.

More input/ideas on specifics are always appreciated.

In half-related news, this is also coming soon:

<img width="614" alt="pwpush-requests" src="https://github.com/user-attachments/assets/de938373-f526-459e-b7ab-d6e5be0ea1c5">

**Comment by riahc3** [comment]
> Hi @riahc3 - I agree/good points.
> 
> I have QR codes planned - you enter the content (text, url, otp code), a QR code is generated and sent to the end user to scan with his/her phone.
> 
> As for certificates - that could be interesting but I would need to know more specifics. Right now agnostic file attachments are the only way.
> 
> More input/ideas on specifics are always appreciated.
> 
> In half-related news, this is also coming soon:
> 
> <img alt="pwpush-requests" width="614" src="https://private-user-images.githubusercontent.com/395132/369844993-de938373-f526-459e-b7ab-d6e5be0ea1c5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjgzODU3MjcsIm5iZiI6MTcyODM4NTQyNywicGF0aCI6Ii8zOTUxMzIvMzY5ODQ0OTkzLWRlOTM4MzczLWY1MjYtNDU5ZS1iN2FiLWQ2ZTViZTBlYTFjNS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDA4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAwOFQxMTAzNDdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iODgxM2UyOWYxMjVkZjhlNTYwZGQzYTRkYmE1OWU0Mzk5NWVlYWZmMDNiNWFjYmZlOWMzZDlkZTkyZjkwODYxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.GMsZHtbMQJJVLqY2A0IWmSauDzv6JertvMR3VzQPopI">

Correct,"agnostic file attachments"; Attach a PFX to the push and send a seperate push as the password for the PFX. Just like a email attachment, etc.

QR Codes will be indeed intresting.

What is the requests tab for?

**Comment by pglombardo** [comment]
> Correct,"agnostic file attachments"; Attach a PFX to the push and send a seperate push as the password for the PFX. Just like a email attachment, etc.

Hi @riahc3 - just to clarify, pwpush does already support agnostic file uploads.  I just want to make sure you didn't miss that feature.

In other news, I've added the ability to send QR codes.  It's currently on the [testing site](https://devlab.pwpush.com/p/new?tab=qr) and will go out to pwpush.com soon.  I'll add it to the OSS version after I wrap up a few in-progress items there.

**Comment by pglombardo** [comment]
HI @riahc3 - QR Pushes have been added in [v1.57.0](https://github.com/pglombardo/PasswordPusher/releases/tag/v1.57.0).  With that and file pushes, I believe that is everything in this issue.

![Image](https://github.com/user-attachments/assets/17734be4-d8b1-4dce-b5c3-66cf8a49232d)

Thanks for the request.  I'll close out this issue but if you have any issues or questions, let me know!
